flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,181 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import date, datetime
+from typing import List, Union, Optional, Literal, NewType
+
+from flowmachine.core.mixins.exposed_datetime_mixin import ExposedDatetimeMixin
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.total_active_periods import (
+    TotalActivePeriodsSubscriber,
+)
+from flowmachine.features.utilities.events_tables_union import EventsTablesUnion
+import dateutil.rrule as rr
+from dateutil.relativedelta import relativedelta
+
+SubscriberSubsetType = NewType("SubscriberSubsetType", Union[str, list, Query, "Table"])
+
+
+class ActiveSubscribers(ExposedDatetimeMixin, Query):
+    """
+    Class that represents subscribers seen to be active.
+
+    To determine this, we regard the timeframe between `start_date` and `end_date` as the major period.
+    We then break the major period into `minor_period_count` minor periods.
+
+    A subscriber is considered to be active in a minor period if they are seen at least `minor_period_
+    threshold` times within that period.
```

Comment:
I think we want to say "seen in at least `minor_period_threshold` distinct 'innermost periods' within that minor period", or something like that,.